golang/testsuite/services_impl/api/api_service.go

```diff
@@ -38,84 +22,14 @@ func NewApiService(serviceCtx *services.ServiceContext, port int) *ApiService {
 	return &ApiService{serviceCtx: serviceCtx, port: port}
 }
 
-// ===========================================================================================
-//                              Service interface methods
-// ===========================================================================================
-func (service ApiService) IsAvailable() bool {
-	url := fmt.Sprintf("http://%v:%v/%v", service.serviceCtx.GetIPAddress(), service.port, healthcheckUrlSlug)
-	resp, err := http.Get(url)
-	if err != nil {
-		logrus.Debugf("An HTTP error occurred when polliong the health endpoint: %v", err)
-		return false
-	}
-	if resp.StatusCode != http.StatusOK {
-		logrus.Debugf("Received non-OK status code: %v", resp.StatusCode)
-		return false
-	}
-
-	body := resp.Body
-	defer body.Close()
-
-	bodyBytes, err := ioutil.ReadAll(body)
-	if err != nil {
-		logrus.Debugf("An error occurred reading the response body: %v", err)
-		return false
-	}
-	bodyStr := string(bodyBytes)
-
-	return bodyStr == healthyValue
-}
-
-// ===========================================================================================
-//                         API service-specific methods
-// ===========================================================================================
-func (service ApiService) getPersonUrlForId(id int) string {
-	return fmt.Sprintf("http://%v:%v/%v/%v", service.serviceCtx.GetIPAddress(), service.port, personEndpoint, id)
+func (service ApiService) GetServiceContext() *services.ServiceContext {
+	return service.serviceCtx
 }
 
-func (service ApiService) AddPerson(id int) error {
-	url := service.getPersonUrlForId(id)
-	resp, err := http.Post(url, textContentType, nil)
-	if err != nil {
-		return stacktrace.Propagate(err, "An error occurred making the request to add person with ID '%v'", id)
-	}
-	if resp.StatusCode != http.StatusOK {
-		return stacktrace.NewError("Adding person with ID '%v' returned non-OK status code %v", id, resp.StatusCode)
-	}
-	return nil
-}
-
-func (service ApiService) GetPerson(id int) (Person, error) {
-	url := service.getPersonUrlForId(id)
-	resp, err := http.Get(url)
-	if err != nil {
-		return Person{}, stacktrace.Propagate(err, "An error occurred making the request to get person with ID '%v'", id)
-	}
-	if resp.StatusCode != http.StatusOK {
-		return Person{}, stacktrace.NewError("Getting person with ID '%v' returned non-OK status code %v", id, resp.StatusCode)
-	}
-	body := resp.Body
-	defer body.Close()
-	bodyBytes, err := ioutil.ReadAll(body)
-	if err != nil {
-		return Person{}, stacktrace.Propagate(err, "An error occurred reading the response body")
-	}
-
-	var person Person
-	if err := json.Unmarshal(bodyBytes, &person); err != nil {
-		return Person{}, stacktrace.Propagate(err, "An error occurred deserializing the Person JSON")
-	}
-	return person, nil
+func (service ApiService) IsAvailable() bool {
+	return true
```

Comment:
Can we make this `false`, so that anyone using the old way always gets an error message? Would help find any accidental uses of the old way

Reply:
Good idea!